### DISCUSSION TITLE
IP 归属地不准导致被误拦截请求

### DISCUSSION THREAD

**Discussion by lainbo** [1Panel-dev/1Panel#4882] [Q&A]
### 1Panel 版本

1.10.6

### 问题描述

发现一个 ip 地址 38.180.150.159，在一些 ip 查询软件中，被认为是荷兰 ip，比如 1Panel 内部，[https://ip-api.com](https://ip-api.com/#38.180.150.159)  、[https://ipapi.is/](https://api.ipapi.is/?q=38.180.150.159)

另外还有一些 ip 查询中，这个被认为是香港 ip，比如[https://ipinfo.io/](https://ipinfo.io/38.180.150.159)  、[https://www.abuseipdb.com/](https://www.abuseipdb.com/check/38.180.150.159)

在一些用户使用代理工具访问我的服务时，他们是香港ip，但是我禁止了荷兰访问，他们就会被误拦截，希望使用更准确的 ip 归属地查询。（根据我自己的经验，ipinfo.io 的准确率更高，仅供参考）

### 重现步骤

1. 高级功能-WAF
2. 全局设置-自定义规则-地区访问限制

这里的设置会导致一些ip被错误的拦截，原因可能是根据ip识别地区错误

**Comment by wanghe-fit2cloud** [comment]
可能和系统使用到的开源IP库有关系，我们会在后续版本考虑优化上述问题。

**Reply by lainbo** [reply]
可以参考一些开源的geoip
https://github.com/MetaCubeX/meta-rules-dat

看有没有帮助

**Reply by wanghe-fit2cloud** [reply]
收到。
